Tests/Tests/ABTesting/ExPlatTests.swift

```diff
@@ -2,18 +2,31 @@ import XCTest
 
 #if SWIFT_PACKAGE
 @testable import AutomatticExperiments
+@testable import AutomatticTracksEvents
 #else
 @testable import AutomatticTracks
 #endif
 
 class ExPlatTests: XCTestCase {
+
     var exPlatTestConfiguration = ExPlatConfiguration(
         platform: "wpios_test",
         oAuthToken: nil,
         userAgent: nil,
         anonId: nil
     )
 
+    var tracksService: TracksService!
+
+    override func setUp() {
+        let contextManager = MockTracksContextManager()
+        self.tracksService = TracksService(contextManager: contextManager)
+    }
+
+    override func tearDown() {
+        ExPlat.shared = nil
+    }
```

Comment:
Have you considered constraining the lifecycle of `trackService` to the test methods?

I consider this approach advantageous because:

1. When tests have a `setUp` method, it's easy to miss the information when looking at a single test
2. Because of the odd way XCTest works, that `tracksService` instance will be retained in memory _until the end of the entire test suite_, unless nilled in the `tearDown`.

I did something similar [in the Parsely iOS library](https://github.com/Parsely/AnalyticsSDK-iOS/pull/78/files#r1172093174) recently.

```swift
func makeTrackService() -> TracksService {
    let contextManager = MockTracksContextManager()
    let tracksService = TracksService(contextManager: contextManager)

    addTeardownBlock {
        ExPlat.shared = nil
    }
}

func testTracksServiceAnonymousConfiguration() {
    let trackService = makeTrackService()
    // ...
```